client/src/main/python/slipstream/executors/node/NodeDeploymentExecutor.py

```diff
@@ -231,24 +231,32 @@ def _run_target_script(self, target_script, exports={}):
 
         process = self._launch_process(target_script, exports)
 
-        # The process is still working on the background.
-        while process.poll() is None:
-            # Ask server whether the abort flag is set. If so, kill the
-            # process and exit. Otherwise, sleep for some time.
-            if self.wrapper.isAbort():
-                try:
-                    util.printDetail('Abort flag detected. '
-                                     'Terminating target script execution...')
-                    process.terminate()
-                    time.sleep(5)
-                    if process.poll() is None:
-                        util.printDetail('Termination is taking too long. '
-                                         'Killing the target script...')
-                        process.kill()
-                except OSError:
-                    pass
-                break
-            time.sleep(self.TARGET_POLL_INTERVAL)
+        try:
+            # The process is still working on the background.
+            while process.poll() is None:
+                # Ask server whether the abort flag is set. If so, kill the
+                # process and exit. Otherwise, sleep for some time.
+                if self.wrapper.isAbort():
+                    try:
+                        util.printDetail('Abort flag detected. '
+                                         'Terminating target script execution...')
+                        process.terminate()
+                        util.sleep(5)
+                        if process.poll() is None:
+                            util.printDetail('Termination is taking too long. '
+                                             'Killing the target script...')
+                            process.kill()
+                    except OSError:
+                        pass
+                    break
+                util.sleep(self.TARGET_POLL_INTERVAL)
+        except IOError as e:
+            if e.errno != errno.EINTR:
+                raise
+            else:
+                util.printDetail('Signal EINTR detected. Ignoring it.')
+                return 0
```

Comment:
Yes. 
`EINTR` happens on Windows when the Python program has some threads or some forks and then Windows is asked to restart. 
In the case of an `EINTR` signal the `process.returncode` will never be 0.
But if it's something else than 0 the Run will fail and we want to be able to restart a machine during the deployment.